### DISCUSSION TITLE
Support Angular @Input alias with argsToTemplate

### DISCUSSION THREAD

**Discussion by bmayen** [storybookjs/storybook#28577] [Ideas]
### Is your feature request related to a problem? Please describe.

Using argsToTemplate with an Angular Material component like Button Toggle and trying to bind and aliased input such as `@Input('aria-label') ariaLabel: string`. 

### Describe the solution you'd like

argsToTemplate should be able to support output `[aria-label]="ariaLabel"` but instead we currently get `[aria-label]="aria-label"` with results in NaN value assigned to the aria-label attribute.

### Are you able to assist to bring the feature to reality?

no

**Comment by Marklb** [comment]
This seems to be a bug in the internal `computesTemplateFromComponent` also. It isn't the fact that it is an alias. The problem appears to be caused by the property name having a special character in it.

I just did a quick test of a fix, which seems to work. I will try to add a more thorough test later and submit a PR, if it works. For a property with special characters, it seems that it needs to be `[aria-label]="this['aria-label']"`.

**Reply by Marklb** [reply]
Submitted a PR that should fix this: https://github.com/storybookjs/storybook/pull/28588

**Reply by bmayen** [reply]
Fantastic. Thank you!
